<launch>
    <arg name="namespace" default="/" />

    <arg name="map_frame" default="map" />
    <arg name="robot_frame" default="base_footprint" />

    <arg name="init_x" default="0.0" />
    <arg name="init_y" default="0.0" />
    <arg name="init_yaw" default="0.0" />

    <arg name="init_pos_err" default="0.15" />
    <arg name="init_yaw_err" default="0.1" />

    <arg name="goal_x" default="0.0" />
    <arg name="goal_y" default="0.0" />
    <arg name="goal_yaw" default="0.0" />

    <arg name="use_sim_time" default="false" />
    
    <arg name="localiser" default="amcl" />
    <arg name="navigator" default="bt_navigator" />

    <arg name="clear_costmaps" default="false" />

    <node name="init_and_goal" pkg="nav2_oneshot_nodes" exec="init_and_goal" output="screen" namespace="$(var namespace)">
        <param name="map_frame" value="$(var map_frame)" />
        <param name="init_x" value="$(var init_x)" />
        <param name="init_y" value="$(var init_y)" />
        <param name="init_yaw" value="$(var init_yaw)" />
        <param name="goal_x" value="$(var goal_x)" />
        <param name="goal_y" value="$(var goal_y)" />
        <param name="goal_yaw" value="$(var goal_yaw)" />
        <param name="init_pos_err" value="$(var init_pos_err)" />
        <param name="init_yaw_err" value="$(var init_yaw_err)" />
        <param name="use_sim_time" value="$(var use_sim_time)" />
        <param name="localiser" value="$(var localiser)" />
        <param name="navigator" value="$(var navigator)" />
        <param name="clear_costmaps" value="$(var clear_costmaps)" />

        <remap from="init_pose/cov" to="initialpose" />
        <remap from="goal_status" to="navigate_to_pose/_action/status" />
    </node>
</launch>